charts/argo-workflows/templates/controller/agent-rb.yaml

```diff
@@ -0,0 +1,29 @@
+{{- if .Values.workflow.rbac.agentPermissions -}}
+  {{- range $namespace := or .Values.singleNamespace false | ternary (list "") (append .Values.controller.workflowNamespaces (coalesce .Values.workflow.namespace (include "argo-workflows.namespace" .)) | uniq)  }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: {{ template "argo-workflows.fullname" $ }}-workflow-agent
+  labels:
+    {{- include "argo-workflows.labels" (dict "context" $ "component" $.Values.controller.name "name" $.Values.controller.name) | nindent 4 }}
+    {{- with $namespace }}
+  namespace: {{ . }}
+    {{- end }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: {{ template "argo-workflows.fullname" $ }}-workflow-agent
+subjects:
+  - kind: ServiceAccount
+    name: {{ $.Values.workflow.serviceAccount.name }}
+    {{- with $namespace }}
+    namespace: {{ . }}
+    {{- end }}
+  {{- range $.Values.workflow.rbac.serviceAccounts }}
```

Comment:
It is similar for the Agent SA, although a bit different, as it is configured by [`spec.serviceAccountName`](https://github.com/argoproj/argo-workflows/blob/1392ef516fd69a2cae875d746c1155f940e4948c/workflow/controller/agent.go#L154) (and at most one Agent runs per Workflow) and so _can_ overlap with other Workflow Pods if those do not have per-template SAs set (i.e. [`spec.templates[].serviceAccountName`](https://argo-workflows.readthedocs.io/en/release-3.5/fields/#template))